compiler/fir/raw-fir/fir-common/src/org/jetbrains/kotlin/fir/builder/BaseFirBuilder.kt

```diff
@@ -732,27 +722,7 @@ abstract class BaseFirBuilder<T>(val baseSession: FirSession, val context: Conte
                             isVararg = false
                         }
                     }
-
-                    // TODO: Handle generic types.
-                    val initCallExpression = buildFunctionCall {
-                        argumentList = buildArgumentList {
-                            for ((ktParameter, firProperty) in zippedParameters) {
-                                val parameterSource = ktParameter?.toFirSourceElement()
-                                arguments += generateComponentAccess(parameterSource, firProperty)
```

Comment:
@demiurg906 also pointed out that this part generates `copy` that looks like:
```
data class Foo(val x: Int, val y: Int) {
    fun copy(x: Int = this.x, y: Int = this.y) = Foo(this.x, this.y)
}
```
which should have been using `Foo(x, y)` instead, since default values for parameters are already set.

Instead of fixing it directly here, I think this entire refactoring/consolidation would make more sense, because now both `psi2ir` and `fir2ir` generate the same body for all the synthetic members for data class.

Reply:
I worried this might indicate the test coverage hole, but it's actually covered by an unmuted test in this PR: https://github.com/JetBrains/kotlin/pull/3300/commits/ccabcb277de2bb7b7afa0c5a1918f343e2204043#diff-a872e759042c15f66b453f6b59f64b7d